src/main/java/org/elasticsearch/common/util/concurrent/PrioritizedEsThreadPoolExecutor.java

```diff
@@ -133,10 +129,13 @@ public Pending(Object task, Priority priority, long insertionOrder, boolean exec
         }
     }
 
-    static class TieBreakingPrioritizedRunnable extends PrioritizedRunnable {
+    private class TieBreakingPrioritizedRunnable extends PrioritizedRunnable {
```

Comment:
minor nit... can you make this class final? 
I also think we should add additional documentation why we are doing all the work here and why we null them out etc. to ensure we don't drop it accidentally. I assume none of the test will fail consistently if we do that so it will be puzzling again